### PR TITLE
fix(modal): focus traps when close button is disabled

### PR DESCRIPTION
**Pull request by anveshmekala**
**Related Issue:** #3064 

 ### Summary
 
 This fix will trap the focus for modal with `disable-close-button`

### CODE REVIEW

**Review by jcfranco — CHANGES_REQUESTED** [review]
Great job adding the test! 

Focus trapping is already set up, and this is adding extra code. Can you rework the solution to leverage [`focusFirstElement`](https://github.com/Esri/calcite-components/blob/master/src/components/modal/modal.tsx#L385-L387). This is where the issue lies.

**Review by jcfranco — APPROVED** [review]
🎉 🌮 🦖 🦃

File: src/components/modal/modal.tsx [hunk 1]
```diff
@@ -304,7 +304,6 @@ export class Modal implements ConditionalSlotComponent {
   @Method()
   async setFocus(focusId?: "close-button"): Promise<void> {
     const closeButton = this.closeButtonEl;
-
```
Comment (jcfranco) [inline src/components/modal/modal.tsx:307]: Sorry, I missed this. Can you restore this? It's an unrelated change.

Comment (anveshmekala) [reply src/components/modal/modal.tsx:307]: ahh, my bad. will restore this.
